### DISCUSSION TITLE
Pagination with large Tables always running the query?

### DISCUSSION THREAD

**Discussion by ChrisSantiago82** [livewire/livewire#1018] [4. General]
Hello,
First I want to congratulate you for this fantastic product. I'm still a beginner but I already love it.

I have a question about the Pagination. I have read the documentation (https://laravel-livewire.com/docs/pagination).

I have a big table with data, I want to have some filters at the top, and of course use the pagination so that I don't have to load the whole table. I also have some buttons that the user can open a single entry and see the details.

My question now, in the documentation the query is in the render function. That means every time the user does something, like pressing another button on the page, the query would run over this large table? I would prefer that it would only run if any of the table filters would change.
Put since the pagination isn't a supported java script type, I can't make a function that is only called when the filter is updated.
What are your thoughts or maybe anybody already has solved this problem.

**Comment by JayBizzle** [comment]
This is pretty much the way Livewire works, not much you can do about it. There query is going to run for each page you navigate to.

**Comment by ChrisSantiago82** [comment]
Thank you for your response. I could live with the fact that it would run everytime I go to another page.
But you must image, it would run on every button click or every time a key is pressed if there is an input field.
And this problem only exists if there is pagination in place, what usually only happens if there is a big table.
There should be a better way to get around this.
And I'm really happy with this program. Just I think in this particular place it could be improved.
